examples/cpp/fill-grid-pp2hadron.cpp

```diff
@@ -0,0 +1,245 @@
+/// Example used in the PineAPPL v1 paper ///
+#include <cstdint>
+#include <pineappl_capi.h>
+
+#include <cassert>
+#include <cmath>
+#include <cstddef>
+#include <random>
+#include <string>
+#include <vector>
+
+struct Psp2to2Hadron {
+    double s;
+    double t;
+    double u;
+    double x1;
+    double x2;
+    double z;
+    double pt_hadron;
+    double y_hadron;
+    double jacobian;
+};
+
+double me_gg2qqbar(double s, double t, double u) {
+    (void) s; // ignore dummy variable
+    double as2 = 0.118 * 0.118;
+    double PI2 = M_PI * M_PI;
+    // TODO: double-check
+    return (16 * PI2 * as2 / 6.0) * (std::pow(u, 2) + std::pow(t, 2)) / (u * t);
```

Comment:
Don't use `std::pow(a, b)` for small integer arguments `b`; this will result in slow code as it's basically equivalent to `std::exp(std::log(a) * b)`. Use `u * u` and `t * t` instead, that's also shorter to write.

Reply:
Yes, I knew. Still, it was the OCD part of me 🙈